Formula/kamel.rb

```diff
@@ -0,0 +1,38 @@
+class Kamel < Formula
+  desc "Apache Camel K CLI"
+  homepage "https://camel.apache.org/"
+
+  url "https://github.com/apache/camel-k.git",
+    :tag      => "1.0.0-RC2",
+    :revision => "24ddce5afb41c70a0bf30e06f74c6f380f5fd851"
+  head "https://github.com/apache/camel-k.git"
+
+  bottle :unneeded
+
+  depends_on "go"
+  depends_on "openjdk"
+
+  def install
+    # Build from source & Install
+    system "make"
+    bin.install "kamel"
+
+    # Install bash completion
+    output = Utils.popen_read("#{bin}/kamel completion bash")
+    (bash_completion/"kamel").write output
+
+    prefix.install_metafiles
+
+    # Install zsh completion
+    output = Utils.popen_read("#{bin}/kamel completion zsh")
+    (zsh_completion/"_kamel").write output
+  end
+
+  test do
+    run_output = shell_output("#{bin}/kamel 2>&1")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I have improved over testing introducing some more cases. The CLI would require a Kubernetes cluster _(or minikube)_ to be fully tested and fails in the same category as _kubernetes-cli_ formula etc.

Reply:
Please run a command and check for an expected error then.

Reply:
add failure testing, if you please review